react/features/base/devices/actions.js

```diff
@@ -0,0 +1,79 @@
+/* globals APP */
+
+import UIEvents from '../../../../service/UI/UIEvents';
+import {
+    SET_AUDIO_INPUT_DEVICE,
+    SET_AUDIO_OUTPUT_DEVICE,
+    SET_VIDEO_INPUT_DEVICE,
+    UPDATE_DEVICE_LIST
+} from './actionTypes';
+
+/**
+ * Signals to update the currently used audio input device.
+ *
+ * @param {string} deviceId - The id of the new audio input device.
+ * @returns {{
+ *      type: SET_AUDIO_INPUT_DEVICE,
+ *      deviceId: string
+ * }}
+ */
+export function setAudioInputDevice(deviceId) {
+    APP.UI.emitEvent(UIEvents.AUDIO_DEVICE_CHANGED, deviceId);
+
+    return {
+        type: SET_AUDIO_INPUT_DEVICE,
+        deviceId
+    };
+}
+
+/**
+ * Signals to update the currently used audio output device.
+ *
+ * @param {string} deviceId - The id of the new audio ouput device.
+ * @returns {{
+ *      type: SET_AUDIO_OUTPUT_DEVICE,
+ *      deviceId: string
+ * }}
+ */
+export function setAudioOutputDevice(deviceId) {
+    APP.UI.emitEvent(UIEvents.AUDIO_OUTPUT_DEVICE_CHANGED, deviceId);
+
+    return {
+        type: SET_AUDIO_OUTPUT_DEVICE,
+        deviceId
+    };
+}
+
+/**
+ * Signals to update the currently used video input device.
+ *
+ * @param {string} deviceId - The id of the new video input device.
+ * @returns {{
+ *      type: SET_VIDEO_INPUT_DEVICE,
+ *      deviceId: string
+ * }}
+ */
+export function setVideoInputDevice(deviceId) {
+    APP.UI.emitEvent(UIEvents.VIDEO_DEVICE_CHANGED, deviceId);
```

Comment:
It didn't even cross my mind. I've created middleware for this now, which would likely be used anyway when the actual device switching is brought into the web react app.